napari/viewer.py

```diff
@@ -54,10 +54,16 @@ def __init__(
             order=order,
             axis_labels=axis_labels,
         )
+        # we delay initialization of plugin system to the first instantiation
+        # of a viewer... rather than just on import of plugins module
+        from .plugins import _initialize_plugins
+
```

Comment:
Maybe run in a separate thread? And update menus after the calculation is done?

Reply:
another pr can tackle adding threading to plugin initialization